flink-kubernetes/src/main/java/org/apache/flink/kubernetes/KubernetesClusterDescriptor.java

```diff
@@ -168,6 +185,13 @@ public String getClusterDescription() {
 		final String nameSpace = flinkConfig.getString(KubernetesConfigOptions.NAMESPACE);
 		flinkConfig.setString(JobManagerOptions.ADDRESS, clusterId + "." + nameSpace);
 
+		if (HighAvailabilityMode.isHighAvailabilityModeActivated(flinkConfig)) {
+			flinkConfig.setString(HighAvailabilityOptions.HA_CLUSTER_ID, clusterId);
```

Comment:
What's the config overriding for?

Reply:
We should update the `HA_CLUSTER_ID` since it will be used to contruct the zk path and ha storage path for the Flink cluster.
Yarn has the similar behavior in `FlinkYarnSessionCli#applyCommandLineOptionsToConfiguration`.

Reply:
So this change is not related to this commit?

Reply:
It is necessary when high-availability enabled. So do you mean to separate it to another commit?

Reply:
Yes. If it fixes another issue, we should make it a different commit.